src/utils.ts

```diff
@@ -124,4 +90,38 @@ export function untilde(str: string) {
     }
     return sub;
   });
-};
\ No newline at end of file
+};
+
+export function retrievePossibleRef(data: any & { $ref?: string }, pathOfData: string, spec: any = {}): any {
+  if (!hasRef(data)) {
+    return data;
+  }
+
+  const refPath = serializePath(data.$ref);
+  if (pathOfData.startsWith(refPath)) { // starts by given path
+    return retrieveDeepData(spec, splitPath(refPath)) || data;
+  } else if (pathOfData.includes(refPath)) { // circular path in substring of path
```

Comment:
Correct me if I'm wrong but given:

`pathOfData` = `/components/channels/test`
`refPath` = `/channels/test`

Would it  match `(pathOfData.includes(refPath)` when it shouldn't?.

Reply:
The `string.includes` in JS can be used to check if a string has a substring.

Reply:
I know. What I want to raise here is that `'/components/channels/test'.includes('/channels/test')` returns `true` **but** I think it is not an expected behavior here since both are completely different paths.